apps/backend/src/lib/rds.ts

```diff
@@ -22,6 +31,106 @@ export class RDS {
             .then((xs) => xs[0]?.id ?? null);
     }
 
+    /**
+     * Retrieves an account with the specified user ID and account type 'GOOGLE'.
+     *
+     * @param db - The database or transaction object.
+     * @param userId - The ID of the user whose account is to be retrieved.
+     * @returns A promise that resolves to the account object if found, otherwise null.
+     */
+    static async getAccountByProviderId(
+        db: DatabaseOrTransaction,
+        accountType: 'GOOGLE' | 'GUEST',
+        providerId: string
+    ): Promise<Account | null> {
+        return db.transaction((tx) =>
+            tx
+                .select()
+                .from(accounts)
+                .where(and(eq(accounts.accountType, accountType), eq(accounts.providerAccountId, providerId)))
+                .limit(1)
+                .then((res) => res[0] ?? null)
+        );
+    }
+
+    static async getAccountUserByToken(db: DatabaseOrTransaction, refreshToken: string) {
+        return db.transaction((tx) =>
+            tx
+                .select()
+                .from(sessions)
+                .innerJoin(users, eq(sessions.userId, users.id))
+                .innerJoin(accounts, eq(users.id, accounts.userId))
+                .where(eq(sessions.refreshToken, refreshToken))
+                .execute()
+                .then((res) => {
+                    return { users: res[0].users, accounts: res[0].accounts };
+                })
+        );
+    }
+
+    /**
+     * Creates a new user and an associated account with the specified provider ID.
+     *
+     * @param db - The database or transaction object.
+     * @param providerId - The provider account ID for the new account.
+     * @returns A promise that resolves to the newly created account object.
+     */
+    static async registerUserAccount(
+        db: DatabaseOrTransaction,
+        providerId: string,
+        name: string,
+        accountType: 'GOOGLE' | 'GUEST',
+        email?: string,
+        avatar?: string
+    ) {
+        const existingAccount = await this.getAccountByProviderId(db, accountType, providerId);
+        if (!existingAccount) {
+            const userId = crypto.randomUUID();
+            await db.insert(users).values({
+                id: userId,
+                avatar: avatar ?? '',
+                name: name,
+                email: email ?? '',
+            });
```

Comment:
You don't need to generate a user ID here. There's already a default for that column in `users`.


```suggestion
            const { userId } = await db.insert(users).values({
                id: userId,
                avatar: avatar ?? '',
                name: name,
                email: email ?? '',
            }).returning({userId: users.id});
```